pennylane/tape/jacobian_tape.py

```diff
@@ -533,7 +538,8 @@ def jacobian(self, device, params=None, **options):
                 # First order (forward) finite-difference will be performed.
                 # Compute the value of the tape at the current parameters here. This ensures
                 # this computation is only performed once, for all parameters.
-                options["y0"] = np.asarray(self.execute_device(params, device))
+                params_f64 = np.array(params, dtype=np.float64)
```

Comment:
One of the jax device tests doesn't seem to like this line :thinking: 
https://github.com/PennyLaneAI/pennylane/pull/1349/checks?check_run_id=2697038221#step:10:63

On another note: wouldn't this casting be a bit too specific at a general place? This would potentially cast the parameters all the time regardless of the interface. Would we only like to do it for the jax interface?

Reply:
@albi3ro I think I agree with @antalszava in this case. While there is a good argument for enforcing finite difference to always use float64, I feel like this should be an interface level option/default, rather than enforced in the Jacobian method